src/tests/ftest/launch.py

```diff
@@ -2088,9 +2109,80 @@ def run_tests(self, sparse, fail_fast, extra_yaml, stop_daos, archive, rename, j
+            command = ["rm", "-fr", BULLSEYE_FILE]
+            if not run_remote(logger, self.bullseye_hosts, " ".join(command)).passed:
+                message = "Error removing bullseye code coverage file on at least one host"
+                self._fail_test(self.result.tests[0], "Run", message, None)
+                return 128
+
+            logger.debug("Copying %s bullseye code coverage source file", BULLSEYE_SRC)
+            command = ["cp", BULLSEYE_SRC, BULLSEYE_FILE]
+            if not run_remote(logger, self.bullseye_hosts, " ".join(command)).passed:
+                message = "Error copying bullseye code coverage file on at least one host"
+                self._fail_test(self.result.tests[0], "Run", message, None)
+                return 128
+
+            logger.debug("Updating %s bullseye code coverage file permissions", BULLSEYE_FILE)
+            command = ["chmod", "777", BULLSEYE_FILE]
+            if not run_remote(logger, self.bullseye_hosts, " ".join(command)).passed:
+                message = "Error updating bullseye code coverage file on at least one host"
+                self._fail_test(self.result.tests[0], "Run", message, None)
+                return 128
+        return 0
+
+    def finalize_bullseye(self):
+        """Retrieve the bullseye code coverage collection information from the hosts.
+
+        Returns:
+            int: status code: 0 = success, 16 = failure
+
+        """
+        if not self.bullseye_hosts:
+            return 0
+
+        bullseye_path, bullseye_file = os.path.split(BULLSEYE_FILE)
+        bullseye_dir = os.path.join(self.job_results_dir, "bullseye_coverage_logs")
+        status = self._archive_files(
+                "bullseye coverage log files", self.bullseye_hosts, bullseye_path,
+                "".join([bullseye_file, "*"]), bullseye_dir, 1, None, 900)
+        # Rename bullseye_coverage_logs.host/test.cov.* to
+        # bullseye_coverage_logs/test.host.cov.*
+        for item in os.listdir(self.job_results_dir):
+            item_full = os.path.join(self.job_results_dir, item)
+            if os.path.isdir(item_full) and "bullseye_coverage_logs" in item:
+                host_ext = os.path.splitext(item)
+                if len(host_ext) > 1:
+                    if not os.path.exists(bullseye_dir):
+                        os.makedirs(bullseye_dir)
+                    for name in os.listdir(item_full):
+                        old_file = os.path.join(item_full, name)
+                        if os.path.isfile(old_file):
+                            new_name = name.split(".")
+                            new_name.insert(1, host_ext[-1])
+                            new_file_name = ".".join(new_name).replace("..", ".")
+                            new_file = os.path.join(bullseye_dir, new_file_name)
+                            logger.debug("Renaming %s to %s", old_file, new_file)
+                            os.rename(old_file, new_file)
```

Comment:
What does this rename do? Do you have an example of old -> new?

Reply:
(1)Current archive the coverage-log under each test-platform (vm, hardware..) sub-directory
ie: under Functional Hardware Large:
     bullseye_coverage_logs.wolf-141
     bullseye_coverage_logs.wolf-142
     bullseye_coverage_logs.wolf-143
...
reference this PR #22
(2)These codes move all the bullseye related logs to a subdirectory:
     bullseye_covreage_log\.

Reply:
Those likes for #22 and #25 appear to be unrelated to this PR. What am I missing?

Reply:
Thanks. So it's renaming, for example:
```
bullseye_coverage_logs.wolf-142/test.cov.bz2 ->
bullseye_coverage_logs/test.wolf-142.cov.bz2
```
If you need to push again, will you please add a comment with an example like this? I think it would help if we need to adjust/debug this code in the future

Reply:
The comment already added:
2166         # Rename bullseye_coverage_logs.host/test.cov.* to
2167         # bullseye_coverage_logs/test.host.cov.*

Reply:
Wow, you're right! Somehow github wasn't rendering all the diff in this view...